src/index.ts

```diff
@@ -245,21 +453,20 @@ if (process.execArgv.find(str => str.startsWith('--inspect-brk='))) {
 				catch (err) { console.error(`delayed 'wait' call failed: ${err.message}`); }
 			}
 		}
-	}
+	};
 
 	wait = <T>(arg: Promise<T> | Thunk<T>): T => {
 		if (isDebugEval()) {
 			if (!fibers.current.delayed) fibers.current.delayed = [];
 			fibers.current.delayed.push(arg);
-			throw 'would yield';
-		} else {
-			flushDelayed();
-			return oldWait(arg);
+			throw new Error('would yield');
 		}
+		flushDelayed();
+		return oldWait(arg);
 	};
 	const oldRun = run;
 	run = <T>(fn: () => T): Promise<T> => {
-		if (isDebugEval()) throw 'would start a fiber';
+		if (isDebugEval()) throw new Error('would start a fiber');
```

Comment:
This code is only executed by the debugger, when it evaluates expressions that call run or wait. It prevents the debugger from freezing on a fiber call. Did you check that it still works with an Error object? 

I think I threw a string rather than an Error object because I did not want to clutter the debugger with error objects. There was also e a memory issue here: debugger allocating a lot of Error objects and stack trace captures vs. a string literal which does not require any dynamic memory allocation.

Reply:
I never experienced debugger issue with f-promise. May be I missed something to test. If you have a specific case, you should check it. For instance I ran example/disk-usage.ts and debug it with chrome inspector without problem.